File: examples/dart/todo/src/wip.rs
#[rid::model]
#[rid::structs(Todo)]
#[derive(Debug, rid::Debug)]
pub struct Model {
    todos: Vec<Todo>,
}

#[rid::model]
#[derive(Debug, rid::Debug)]
pub struct Todo {
    id: u32,
    title: String,
}

#[rid::export]
impl Model {
    #[rid::export(initModel)]
    fn new() -> Model {
        Self {
            todos: vec![
                Todo {
                    id: 0,
                    title: "first todo".to_string(),
                },
                Todo {
                    id: 1,
                    title: "second todo".to_string(),
                },
            ],
        }
    }

    #[rid::export]
    #[rid::structs(Todo)]
    fn filtered_todos(&self) -> Vec<&Todo> {
        self.todos.iter().filter(|x| x.id > 0).collect()
    }
}

File: examples/dart/todo/lib/wip.dart
import 'generated/ffigen_binding.dart';
import 'generated/rid_generated.dart';
import 'dart:io';
import 'dart:async';

import 'log.dart';

printFiltered(Pointer<Model> model) {
  final filtered = model.filtered_todos();
  final list = filtered.iter().toList();
  filtered.dispose();
  log.i("\n${list.first.debug(true)}");
}

main() {
  final model = rid_ffi.initModel();
  log.i("\n${model.debug(true)}");
  printFiltered(model);
  printFiltered(model);
  model.dispose();
}
